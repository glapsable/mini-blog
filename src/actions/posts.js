export const addPost = (postData = {}) => {
  const {
    title = '',
    createdAt = 0,
    amount = 0,
    description = '',
  } = postData;
  const post = {
    title, createdAt, amount, description,
  };
  return {
    type: 'ADD_POST',
    post,
  };
};

export const removePost = ({ id } = {}) => ({
  type: 'REMOVE_POST',
  id,
});

export const editPost = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates,
});
