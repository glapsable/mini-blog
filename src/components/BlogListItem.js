import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import moment from 'moment';

// load a locale
numeral.register('locale', 'pl', {
  delimiters: {
    thousands: ' ',
    decimal: ',',
  },
  abbreviations: {
    thousand: 'k',
    million: 'm',
    billion: 'b',
    trillion: 't',
  },
  currency: {
    symbol: 'PLN',
  },
});

// switch between locales
numeral.locale('pl');

const BlogListItem = ({
  id, title, amount, createdAt, description,
}) => (
  <Link to={`/edit/${id}`}>
    <h3>{title}</h3>
    <p>{description}</p>
    <p>Trip amount: {numeral(amount / 100).format('0,0[.]00 $')}</p>
    <p>{moment(createdAt).format('Do MMMM YYYY')}</p>
  </Link>
);

BlogListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  createdAt: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
};

export default BlogListItem;
