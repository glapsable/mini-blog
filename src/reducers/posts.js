const postsReducerDefaultState = [];
export default (state = postsReducerDefaultState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
