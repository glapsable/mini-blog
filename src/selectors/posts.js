import moment from 'moment';

export default (posts, {
  text, sortBy, startDate, endDate,
}) => posts.filter((post) => {
  const createdAtMoment = moment(post.createdAt);
  const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
  const endDateMatch = endDate ? endDate.isSameOrBefore(createdAtMoment, 'day') : true;
  const textMatch = post.title.toLowerCase().includes(text.toLowerCase());

  return startDateMatch && endDateMatch && textMatch;
}).sort((a, b) => {
  if (sortBy === 'date') return a.createdAt < b.createdAt ? 1 : -1;
  if (sortBy === 'amount') return a.amount < b.amount ? 1 : -1;
  return 0;
});
