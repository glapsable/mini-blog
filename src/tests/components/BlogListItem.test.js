import React from 'react';
import { shallow } from 'enzyme';
import BlogListItem from '../../components/BlogListItem';
import posts from '../fixtures/posts';

test('should render BlogListItem with given post correctly', () => {
  const wrapper = shallow(<BlogListItem {...posts[2]} />);
  expect(wrapper).toMatchSnapshot();
});
