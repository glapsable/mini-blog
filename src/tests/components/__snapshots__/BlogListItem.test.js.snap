// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BlogListItem with given post correctly 1`] = `
<Link
  replace={false}
  to="/edit/ccc"
>
  <h3>
    ccc
  </h3>
  <p>
    ccc
  </p>
  <p>
    Trip amount: 
    0,50 PLN
  </p>
  <p>
    8th January 1970
  </p>
</Link>
`;
